core/wiki/macros/tree.tid

```diff
@@ -31,32 +24,34 @@ tags: $:/tags/Macro
 </$button>
 </$reveal>
 &#32;
-<span>(<$count filter="[all[shadows+tiddlers]removeprefix<__prefix__>removeprefix<__chunk__>] -[<__prefix__>addsuffix<__chunk__>]"/>)</span>
+<span>(<$count filter="[all[shadows+tiddlers]get<__field__>removeprefix<__prefix__>removeprefix<__chunk__>] -[<__prefix__>addsuffix<__chunk__>]"/>)</span>
 <$reveal type="match" stateTitle=<<reveal-state>> text="show">
-<$macrocall $name="tree-node" prefix={{{ [<__prefix__>addsuffix<__chunk__>] }}} separator=<<__separator__>>/>
+<$macrocall $name="tree-node" prefix={{{ [<__prefix__>addsuffix<__chunk__>] }}} field=<<__field__>> separator=<<__separator__>>/>
 </$reveal>
 </$set>
 </li>
 \end
 
-\define tree-node(prefix,separator: "/")
+\define tree-node(prefix,field,separator: "/")
 \whitespace trim
 <ol>
-<$list filter="[all[shadows+tiddlers]removeprefix<__prefix__>splitbefore<__separator__>sort[]!suffix<__separator__>]" variable="chunk">
-<$macrocall $name="leaf-node" prefix=<<__prefix__>> chunk=<<chunk>> separator=<<__separator__>>/>
+<$list filter="[all[shadows+tiddlers]get<__field__>prefix<__prefix__>removeprefix<__prefix__>splitbefore<__separator__>sort[]!suffix<__separator__>]" variable="chunk">
+<$list filter="[all[shadows+tiddlers]has<__field__>sort[]] :filter[get<__field__>removeprefix<__prefix__>match<chunk>]" variable="title">
+<$macrocall $name="leaf-node" prefix=<<__prefix__>> chunk=<<chunk>> title=<<title>> separator=<<__separator__>>/>
+</$list>
```

Comment:
Having 2 lists that iterate over every shadow and every tiddler starting from new, may be the performance bottle neck. So if possible we may be able to "reuse" some results from the first list as input to the second one. -- **But** I did not have a look at the code at the moment.

Reply:
I agree that the performance must be improved. The challenge, I think, is that we need to (1) find and sort the list of all valid field values (i.e., those with a given prefix) and (2) find all tiddlers with the given field values

One strategy may be to

1. Find tiddlers with valid field values
2. Join field value and tiddler title with the separator
3. Sort the list of field value-tiddler title strings
4. Split by the separator and somehow extract the last element (tiddler title)

But that would require a larger re-write of the tree macro